test/master-playlist-controller.test.js

```diff
@@ -368,6 +368,61 @@ QUnit.test('updates the enabled track when switching audio groups', function(ass
            'enabled a track in the new audio group');
 });
 
+QUnit.test('detects if the player is stuck at the playlist end', function(assert) {
+  let playlistCopy = Hls.Playlist.playlistEnd;
+
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  this.standardXHRResponse(this.requests.shift());
+  let playlist = this.player.tech_.hls.selectPlaylist();
+
+  // not stuck at playlist end when no seekable, even if empty buffer
+  // and positive currentTime
+  this.masterPlaylistController.seekable = () => videojs.createTimeRange();
+  this.player.tech_.buffered = () => videojs.createTimeRange();
+  this.player.tech_.setCurrentTime(170);
+  assert.ok(!this.masterPlaylistController.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when no seekable, even if empty buffer
+  // and currentTime 0
+  this.player.tech_.setCurrentTime(0);
+  assert.ok(!this.masterPlaylistController.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when no seekable but current time is at
+  // the end of the buffered range
+  this.player.tech_.buffered = () => videojs.createTimeRange(0, 170);
+  assert.ok(!this.masterPlaylistController.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
+
+  // not stuck at playlist end when currentTime not at seekable end
+  // even if the buffer is empty
+  this.masterPlaylistController.seekable = () => videojs.createTimeRange(0, 130);
+  this.player.tech_.setCurrentTime(50);
+  this.player.tech_.buffered = () => videojs.createTimeRange();
+  Hls.Playlist.playlistEnd = () => 130;
+  assert.ok(!this.masterPlaylistController.stuckAtPlaylistEnd_(playlist), 'not stuck at playlist end');
```

Comment:
Good point. I think that leaving in tests that make use of seekable values is good, even if `stuckAtPlaylistEnd_` doesn't use them, since the tests shouldn't know too much about the internal workings of the function. But you're right that the scenario of `playlistEnd` equaling `seekable`'s end shouldn't be a common case. However, in the event that a playlist changes from LIVE to VOD (at the end of a stream), and the only update was the addition of `#EXT-X-ENDLIST`, `seekable`'s end should equal `playlistEnd`, and I think it's possible that this case is a valid one.